### DISCUSSION TITLE
Finalize_Model

### DISCUSSION THREAD

**Discussion by rlee300** [pycaret/pycaret#1157] [Q&A] [answered]
My data contains >1.5 millions of rows and 22 columns. Is it reasonable for it take >1 hour to "finalize_model()", when "create_model()" only takes 20 min?

**Comment by Yard1** [comment, marked as answer]
It's possible that you are running out of memory. `finalize_model` fits the entire dataset, while `create_model` does CV - depending on your fold setting, it can mean that between 10% to 25% of the train set is not fitted at a given point in time, and whatever percentage you have chosen for the test set (by default - 30%) - therefore, running out of memory would be my main guess. If you are running it locally it probably moves to swap memory which is orders of magnitude slower.
